src/index.ts

```diff
@@ -1,26 +1,58 @@
-import { Client } from "discord.js";
+import { ApplicationCommandData, Client, Intents, Message, Presence } from "discord.js";
 import { parse } from "dotenv";
 import exitHook from "exit-hook";
 import { readFileSync } from "fs";
 import { connect, connection, disconnect } from "mongoose";
-import { getCommand } from "./commands";
-import { isConfig } from "./config";
+import { getCommand, getCommands } from "./commands";
+import { Config, isConfig } from "./config";
 import { DotEnv, isDotEnv } from "./dotEnv";
-import { ephemeral } from "./ephemeral";
 import type { State } from "./state";
 
 // USAGE: npm start [configPath] [envPath]
 const configPath = process.argv[2] ?? "config.json";
 const envPath = process.argv[3] ?? ".env";
 
+function deploySlashCommands(client: Client, config: Config, message?: Message) {
+	const data = [] as ApplicationCommandData[];
+
+	getCommands().forEach(command => {
+		data.push({
+			name: command.name,
+			description: command.description,
+			options: command.options,
+		} as ApplicationCommandData);
+	});
+
+	data.forEach(commandData => {
+		(async () => {
+			await client.guilds.cache.get(config.guildId)?.commands.create(commandData);
+		})().catch(console.error.bind(console));
+	});
+
+	if (message) {
+		message.reply("Successfully loaded slash-commands.").catch(console.error.bind(console));
+	}
+}
+
 function main(state: State, env: DotEnv) {
 	const { config, client } = state;
 
 	client.on("ready", () => {
-		client.user
-			?.setActivity(state.version, { type: "PLAYING" })
-			.then(presence => console.log(`Activity set to ${presence.activities[0].name}`))
-			.catch(console.error.bind(console));
+		const presence = client.user?.setActivity(state.version, { type: "PLAYING" }) as Presence;
+
+		console.log(`Activity set to ${presence.activities[0].name}`);
```

Comment:
No reason to log the activity change is there?

```suggestion
		client.user?.setActivity(state.version, { type: "PLAYING" });
```